R/best_and_worst_cases.R

```diff
@@ -0,0 +1,23 @@
+best_and_worst_cases <- function(data) {
+  data |>
+    mutate(amount_of_distinct_products = dplyr::n_distinct(.data$ep_product), .by = "companies_id") |>
+    mutate(equal_weight = (1/.data$amount_of_distinct_products)) |>
+    lowest_risk_category_per_company_benchmark(.by = c("companies_id", "benchmark")) |>
+    highest_risk_category_per_company_benchmark(.by = c("companies_id", "benchmark")) |>
+    mutate(dummy_best = ifelse(.data$emission_profile == .data$min_risk_category_per_company_benchmark, 1, 0)) |>
+    mutate(dummy_worst = ifelse(.data$emission_profile == .data$max_risk_category_per_company_benchmark, 1, 0)) |>
+    mutate(count_best_case_products_per_company_benchmark = sum(.data$dummy_best), .by = c("companies_id", "benchmark")) |>
```

Comment:
I really like names like `count_best_case_products_per_company_benchmark` since it makes it easy to understand what that variable holds. I would preserve those names in this function, and if you/Anne/Tilman want to rename them, then do it outside this function -- in a `polish*()` function right before the entire process ends. This function needs to not know anything about the final use-facing names.

Note it's OK to define and mention names like `count_best_case_products_per_company_benchmark` inside this function. That name is something you (the developer) control. 

I think `dummy_` prefix can be removed already.

Reply:
Removed `dummy_`